tests/host_callback_to_tf_test.py

```diff
@@ -46,7 +48,8 @@ def call_tf_no_ad(tf_fun: Callable, arg, *, result_shape):
   JAX staged computation."""
 
   def tf_to_numpy(t):
-    return t.numpy() if isinstance(t, tf.Tensor) else t
+    # Turn the Tensor to NumPy array without copying.
+    return np.array(memoryview(t)) if isinstance(t, tf.Tensor) else t
```

Comment:
I was actually more concerned about copying from GPU -> CPU, but indeed CPU copies are also a concern.

Unfortunately `np.array(memoryview(t))` still makes a copy (`asarray` copies by default). You would need `np.array(memoryview(t), copy=False)` or `np.asarray(memoryview(t))`. The `memoryview()` part is also likely optional, given that tensorflow implements NumPy's `__array__` conversion protocol.

Reply:
Understood. The example that I tried to copy was using `np.asarray` indeed, but I mis-copied it :-) I changed to asarray